runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/cache/KafkaCachePartition.java

```diff
@@ -415,13 +440,25 @@ public void writeEntryContinue(
         final int logRequired = payload.sizeof();
         assert logAvailable >= logRequired;
 
-        logFile.appendBytes(payload.buffer(), payload.offset(), payload.sizeof());
+        if (payload != null)
+        {
+            final ValueConsumer writeValue = (buffer, index, length) ->
+            {
+                logFile.appendBytes(buffer, index, length);
+            };
```

Comment:
```suggestion
            final ValueConsumer writeValue = logFile::appendBytes;
```

Also, this can be hoisted to constructor and reused.